### DISCUSSION TITLE
Step-by-step guide for using imported custom map in ScenarioRunner

### DISCUSSION THREAD

**Discussion by bbiro2** [carla-simulator/carla#4146] [General]
Hello everyone,

I am interested whether there was a step-by-step guide for using imported custom map in ScenarioRunner?

I already found the [guide for the map importing](https://carla.readthedocs.io/en/0.9.7/how_to_make_a_new_map/) (which I already successfully did), and also found the [guide to run default OpenSCENARIO files in ScenarioRunner](https://carla-scenariorunner.readthedocs.io/en/latest/getting_started/#running-scenarios-using-the-openscenario-format) (which I also successfully did).

My problem is that when I try to combine these 2 actions and just use my custom imported map within ScenarioRunner, it somehow does not work (e.g. it does not find the world it requires even if it is referenced in the xosc file,…etc.)

It would be more than appreciated if there was a simple end-to-end example for this process as a whole, but I did not find any.
For example: having an rrscene and a simple scenario description as an input, and then the guide would follow us through the process where the output would be a running scenario (sourced from a simple OpenSCENARIO file created along the lines of the scenario description mentioned above) in ScenarioRunner.

Did anyone find such guideline?

Thank you in advance!

**Comment by germanros1987** [comment]
Is the map registered? In other words, can you load the map through the API by name? If so, ScenarioRunner should work fine. Could you please provide further details?

**Reply by bbiro2** [reply]
Hi, thanks for replying. 

> Is the map registered? In other words, can you load the map through the API by name?

Could you please provide further details which exact command do you mean when talking about "loading the map through the API by name"? Because I have tried several methods, and I am not sure whether yours is one of them or not.

> Could you please provide further details?

Unfortunately I cannot give you an example scenario, but for example, even a guideline to one of the downloadable OpenSCENARIO standard examples would be highly appreciated.
